landoapi/uplift.py

```diff
@@ -42,46 +41,15 @@
 # Maximum number of revisions allowable in a stack to be auto-uplifted.
 MAX_UPLIFT_STACK_SIZE = 5
 
-ARC_DIFF_REV_RE = re.compile(
-    r"^\s*Differential Revision:\s*(?P<phab_url>https?://.+)/D(?P<rev>\d+)\s*$",
-    flags=re.MULTILINE,
-)
-ORIGINAL_DIFF_REV_RE = re.compile(
-    r"^\s*Original Revision:\s*(?P<phab_url>https?://.+)/D(?P<rev>\d+)\s*$",
-    flags=re.MULTILINE,
-)
-
-
-def move_drev_to_original(body: str) -> str:
-    """Handle moving the `Differential Revision` line.
-
-    Moves the `Differential Revision` line to `Original Revision`, if a link
-    to the original revision does not already exist. If the `Original Revision`
-    line does exist, scrub the `Differential Revision` line.
-
-    Args:
-        body: `str` text of the commit message.
-
-    Returns:
-        New commit message body text as `str`.
-    """
-    differential_revision = ARC_DIFF_REV_RE.search(body)
-    original_revision = ORIGINAL_DIFF_REV_RE.search(body)
-
-    # If both match, we already have an `Original Revision` line.
-    if differential_revision and original_revision:
-        return body
-
-    def repl(match):
-        phab_url = match.group("phab_url")
-        rev = match.group("rev")
-        return f"\nOriginal Revision: {phab_url}/D{rev}"
+UPLIFT_BUG_UPDATE_RETRIES = 3
 
-    # Update the commit message.
-    return ARC_DIFF_REV_RE.sub(repl, body)
 
+def add_original_revision_line(summary: str, uri: str) -> str:
+    """Return the summary with `Original Revision` added."""
+    if any(line.startswith("Original Revision:") for line in summary.splitlines()):
```

Comment:
I think this check should be in `create_uplift_revision`. Not sure if this method is necessary altogether, but if you want to keep it it should be renamed to `add_original_revision_line_if_needed`.

Reply:
This function is here mostly so we can have a unit test of the behaviour. I'll update the function name.